src/meta/src/barrier/recovery.rs

```diff
@@ -543,6 +544,10 @@ impl GlobalBarrierManager {
             paused = ?self.state.paused_reason(),
             "recovery success"
         );
+
```

Comment:
Not sure if we should also notify at the beginning of recovery.

Reply:
I think in most cases, notification after recovery success is enough. Theoretically, we don't need to reconstruct the connection pool, but in some corner cases, tonic couldn't reconnect, so reconstruction here is to deal with this case.